### DISCUSSION TITLE
Update intervall

### DISCUSSION THREAD

**Discussion by lynntryanderror** [StephanJoubert/home_assistant_solarman#281] [Q&A]
Hello,

first of all thanks very much for the integration!

I wanted to change the update intervall for pulling the data. Were in the code can i change that?

Kind regards
Lynn

**Comment by gdwa** [comment]
/config/custom_components/solarman/const.py

MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=**120**)

**Reply by lynntryanderror** [reply]
Thanks very much

**Reply by speedster-pta** [reply]
I've changed this to seconds=0 and it was updating at about 30 seconds intervals. I see now it is back to between one and two minutes, although I haven't changed anything. What would cause the delay in updates?

**Reply by PV-Joe** [reply]
Same here. I set it to 30 seconds. I did. work for a while and the update interval for my Deye 8K Hybrid is back to one minute.

I  would still like to speed it up.

Any other solutions to increase the interval?

**Reply by speedster-pta** [reply]
My problem was that for some reason the logger was disconnecting from the wifi intermittently. This has since resolved itself and updates are consistent at 30 secs

**Comment by PV-Joe** [comment]
I checked it and my configuration for the Deye 8K Hybrid inverter was:

MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)

The only entity that is updating with the 30 seconds interval is the production.

**Comment by PV-Joe** [comment]
After a couple of days my Solarman update Intervall for the production is back to 1 minute.

I didn’t change anything and it is still set to 30 seconds.

I don’t know what is going on and don’t understand this behavior 🤔

**Reply by gdwa** [reply]
The stick is not reliable at all. 
I ask for a 2 minutes refresh to give it some relief. When looking at up-time , it's rebooting every 5 to 10 minutes. It's a crap device. It's only a feeling but the less I stress it, the most chances to receive an update I get. 

I keep an eye on the last update time stamp and put some notifications towards my iPhone when no update for 30 min. Very rarely happening. ![image](https://github.com/StephanJoubert/home_assistant_solarman/assets/58270239/9f4124a7-3ee5-4f3f-886f-754ffba78124)

**Comment by PV-Joe** [comment]
![IMG_3536](https://github.com/StephanJoubert/home_assistant_solarman/assets/110498562/d9ae2c66-cd6c-41ab-a6b6-06ff424d250f)

These values are updating every minute and working absolutely reliable since 9 months.
